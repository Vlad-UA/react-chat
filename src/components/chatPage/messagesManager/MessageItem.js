import React from 'react';
import classNames from 'classnames';

import {withStyles} from 'material-ui/styles';
import {ListItem} from 'material-ui/List';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';

import AvatarItem from "../common/Avatar";

const styles = theme => ({
  root:{
    width: '100%',
  },
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    minWidth: '10%',
    maxWidth: '70%',
  }),

  myMessageListItem: {
    display: 'flex',
    flexDirection: 'row-reverse',
  },

  myMessagePaper: {
    backgroundColor: '#C7EDFC',
  },
});

const MessageItem = ({classes, sender, content}) => {
  const isMessageFromMe = sender === "me";

  return (
    <ListItem className={classNames(classes.root, isMessageFromMe && classes.myMessageListItem)}>
      <AvatarItem title={sender} lettersQuantity={1} colorFrom={sender}/>
      <Paper className={classNames(classes.paper, isMessageFromMe && classes.myMessagePaper)} elevation={4}>
        <Typography component="p" variant="caption">
          {sender}
        </Typography>
        <Typography variant="body1" component="p">
          {content}
        </Typography>
      </Paper>
    </ListItem>
  )
};

export default withStyles(styles)(MessageItem);
