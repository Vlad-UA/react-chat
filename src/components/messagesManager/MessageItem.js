import React from 'react';
import {withStyles} from 'material-ui/styles';
import classNames from 'classnames';

import {ListItem} from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';

import titleInitial from '../../utils/title-initials';

const styles = theme => ({
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
  }),

  myMessageListItem: {
    display: 'flex',
    flexDirection: 'row-reverse',
  },

  myMessagePaper: {
    backgroundColor: '#C7EDFC',
  },
});

class MessageItem extends React.Component {
  render() {
    const {classes, message} = this.props;

    const isMessageFromMe = message.sender === "me";

    return (
      <ListItem className={classNames(isMessageFromMe && classes.myMessageListItem)}>
        <Avatar>
          {titleInitial({title: message.sender, lettersQuantity: 1})}
        </Avatar>
        <Paper className={classNames(classes.paper, isMessageFromMe && classes.myMessagePaper)} elevation={4}>
          <Typography component="p">
            {message.sender}
          </Typography>
          <Typography variant="headline" component="p">
            {message.content}
          </Typography>
        </Paper>
      </ListItem>
    )
  }
}

export default withStyles(styles)(MessageItem);
