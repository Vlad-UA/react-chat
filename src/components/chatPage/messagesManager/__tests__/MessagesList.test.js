/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageItem from '../MessagesList';

const mockProps = {
  sender: {},
  content: 'content',
  activeUser: {
    _id: '12345',
    username: 'me',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  messagesList: [
    {
      _id: '123foo321',
      chatId: '12345',
      content: 'Hello, World!',
      sender: {
        _id: '12345',
        username: 'me',
      },
      createdAt: '2018-03-16T10:53:23.200Z',
    },
    {
      _id: '321bar123',
      chatId: '12345',
      content: 'Hello, React!',
      sender: {
        _id: '54321',
        username: 'someone',
      },
      createdAt: '2018-03-16T10:53:23.200Z',
    },
  ],
};

jest.mock('../MessageItem', () => () => 'MessageItem');

describe('<MessagesList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageItem {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly - renders messages', () => {
    const tree = renderer
      .create(<MessageItem {...mockProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - renders without messages', () => {
    const tree = renderer
      .create(<MessageItem {...mockProps} messagesList={[]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
