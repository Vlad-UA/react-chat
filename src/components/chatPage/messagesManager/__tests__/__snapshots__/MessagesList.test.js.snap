// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesList /> snapshot renders correctly - renders messages 1`] = `
<main
  className="MessagesList-root-7"
>
  <ul
    className="MuiList-root-9 MuiList-padding-10"
  >
    MessageItem
    MessageItem
  </ul>
</main>
`;

exports[`<MessagesList /> snapshot renders correctly - renders without messages 1`] = `
<main
  className="MessagesList-root-7"
>
  <div
    className="MessagesList-thereNoMessage-8"
  >
    <h1
      className="MuiTypography-root-13 MuiTypography-display1-17"
    >
      There is no messages yet...
    </h1>
  </div>
</main>
`;
